src/librustc/hir/mod.rs

```diff
@@ -1014,7 +1026,7 @@ pub enum Expr_ {
     /// A closure (for example, `move |a, b, c| {a + b + c}`).
     ///
     /// The final span is the span of the argument block `|...|`
-    ExprClosure(CaptureClause, P<FnDecl>, BodyId, Span),
+    ExprClosure(CaptureClause, P<FnDecl>, BodyId, Span, Option<GeneratorClause>),
```

Comment:
It's worth adding a comment -- what is an `Option<GeneratorClause>`?

(Also, I wonder if we should convert this to a struct variant at some point.)